app/controllers/spree/paybright_controller.rb

```diff
@@ -52,8 +56,17 @@ def handle_callback_params!
         response_code: paybright_params[:x_gateway_reference],
         amount: paybright_params[:x_amount]
       )
-      @payment.complete!
-      advance_and_complete(@payment.order)
+
+      begin
+        @payment.complete!
+        advance_and_complete(@payment.order)
+      rescue StandardError
+        if @payment.response_code.present?
+          @payment.void
```

Comment:
Can you add a test that the payment is cancelled when an error occurs during advancing?

Reply:
yeah i didnt want to add any without first getting approval that this flow is fine from people

Reply:
Cool it makes sense to me 👍